django_project/trail/models/trail.py

```diff
@@ -0,0 +1,118 @@
+
+# coding=utf-8
+"""Model definitions for a trail app"""
+import os
+
+from django.contrib.gis.db import models
+from django.utils.translation import ugettext_lazy as _
+from django.utils.text import slugify
+from django.core.urlresolvers import reverse
+from unidecode import unidecode
+
+from core.settings.contrib import STOP_WORDS
+from django.conf.global_settings import MEDIA_ROOT
+
+__author__ = 'Alison Mukoma <alison@kartoza.com>'
+__date__ = '01/09/2018'
+__license__ = "GPL"
+__copyright__ = 'kartoza.com'
+
+
+def increment_slug(_name, _colour):
+    """Function to increment slug.
+
+    If there already exists the trail name and colour,
+    the slug will be incremented
+    e.g. trail-#25HHfff, trail-#25HHfff etc.
+    """
+
+    existing_trails = Trail.objects.all()
+    new_trail_name = '%s %s' % (_name, _colour)
+    for trail in existing_trails:
+        if _name == trail.name and _colour == trail.colour:
+            _count_names = Trail.objects.filter(
+                name=_name,
+                colour=_colour)
+            count = _count_names.count() + 1
+            new_trail_name = '%s %s %s' % (_name, _colour, count)
+            break
+
+    return new_trail_name
+
+
+class Trail(models.Model):
+    "Model definition of a Trail."
+
+    name = models.CharField(
+        _('Name of Trail'),
+        max_length = 255,
+        null = True,
```

Comment:
null false - this should be required (and in all other models the name should be required)